src/compas/scene/sceneobject.py

```diff
@@ -42,15 +48,46 @@ def __new__(cls, item, **kwargs):
 
     def __init__(self, item, **kwargs):
         self._item = item
-        self._transformation = None
         self._guids = None
+        self._node = None
+        self._frame = kwargs.get("frame", None)
+        self._transformation = kwargs.get("transformation", None)
+        self.name = kwargs.get("name", item.name or item.__class__.__name__)
         self.color = kwargs.get("color", self.color)
         self.opacity = kwargs.get("opacity", 1.0)
 
+    @property
+    def item(self):
+        return self._item
+
     @property
     def guids(self):
         return self._guids or []
 
+    @property
+    def node(self):
+        return self._node
+
+    @property
+    def parent(self):
+        if self.node:
+            return self.node.parent_object
+
+    @property
+    def children(self):
+        if self.node:
+            return self.node.children_objects
+        else:
+            return []
```

Comment:
Adding `parent` and `children` property to `SceneObject`. They are fetched through associated `SceneObjectNode`.